### PR TITLE
refactor(api): crypto project vcs_Server

### PR DESCRIPTION
**Pull request by fsamin**
Signed-off-by: francois  samin <francois.samin@corp.ovh.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13281.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✘  
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13297.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/api/workflow_application.go [hunk 1]
```diff
@@ -72,9 +72,9 @@ func (api *API) releaseApplicationWorkflowHandler() service.Handler {
 			return sdk.WithStack(sdk.ErrNoReposManager)
 		}
 
-		rm := repositoriesmanager.GetProjectVCSServer(*proj, app.VCSServer)
-		if rm == nil {
-			return sdk.WithStack(sdk.ErrNoReposManager)
+		rm, err := repositoriesmanager.LoadProjectVCSServerLinkByProjectKeyAndVCSServerName(ctx, api.mustDB(), key, app.VCSServer)
+		if err != nil {
+			return sdk.WrapError(sdk.ErrNoReposManagerClientAuth, "cannot get client %s %s got: %v", key, app.VCSServer, err)
```
Comment (richardlt) [inline engine/api/workflow_application.go:77]:
```suggestion
			return sdk.NewErrorWithStack(err, sdk.WrapError(sdk.ErrNoReposManagerClientAuth, "cannot get client %s %s", key, app.VCSServer))
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13334.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13335.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Pull request merged**